immscore/immsutil.cc

```diff
@@ -197,3 +199,20 @@ bool file_exists(const string &filename) {
     struct stat buf;
     return !stat(filename.c_str(), &buf);
 }
+
+string error_string(int errnum) {
+    char buffer[1024];
+#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE
```

Comment:
If we need to support the XSI-compliant version, I think it's simpler to just use that version in all cases.

Reply:
We can't because [g++ defines _GNU_SOURCE](https://gcc.gnu.org/onlinedocs/libstdc++/faq.html#faq.predefined), yet the XSI-compliant version is "preferred for portable applications".
I guess the question is: How portable do we want to be?